test/cpp/end2end/end2end_test.cc

```diff
@@ -337,7 +338,11 @@ class End2endTest : public ::testing::TestWithParam<TestScenario> {
     }
   }
 
-  void ResetStub() {
+  bool ResetStub() {
```

Comment:
I don't know if this check belongs in ResetStub. Maybe a macro would be more readable?

`MAYBE_SKIP_TEST_IF_NEEDED` with a comment about why we can only test TCP callback if there is a bg poller